hook/type_test.go

```diff
@@ -0,0 +1,99 @@
+package hook_test
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/buildkite/agent/v3/hook"
+)
+
+type testCase struct {
+	name             string
+	hookPath         []string
+	expectedHookType string
+	errCheck         func(*testing.T, error) bool
+}
+
+func noErr(t *testing.T, err error) bool {
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		return true
+	}
+	return false
+}
+
+func isNotExist(t *testing.T, err error) bool {
+	if !errors.Is(err, os.ErrNotExist) {
+		t.Errorf("expected os.ErrNotExist, got %v", err)
+		return true
+	}
+
+	return false
+}
+
+func TestHookType(t *testing.T) {
+	wd, _ := os.Getwd()
```

Comment:
Add a comment that this is because test/fixtures/hook is at the root of the repo, but the test working directory is not?